### DISCUSSION TITLE
保存课程表时自动检查是否有不同节点的活动时间重叠

### DISCUSSION THREAD

**Discussion by Yersmagit** [Class-Widgets/Class-Widgets#295] [Feature]
### 检查清单

- [X] 我已更新到最新版 ![最新的版本号](https://img.shields.io/github/v/release/RinLit-233-shiroko/Class-Widgets?include_releases&style=flat-square&label=) ，并看过[最新提交](https://github.com/RinLit-233-shiroko/Class-Widgets/commits/main/)，确认我想要的功能还没有实现。
- [X] 我已在 [Issues](https://github.com/RinLit-233-shiroko/Class-Widgets/issues?q=label%3ABug) 中检索，确认这一 Bug 未被提交过。
- [X] 我认为我想要的功能是用户广泛需要的，插件不能替代的，且没有超出 **软件本来** 的开发目标，而非 **添加与课表及教学辅助无关的内容** 。
- [X] 我已知晓由于开发者时间精力有限，我提交的功能请求可能需要比较久的时间才能实现。

### 背景与动机

添加此功能有助于优化软件使用体验

### 想要实现或优化的功能

希望可以在用户点击了课程表编辑页面的保存按钮时，自动检测是否有 **两个相邻节点的开始时间之差 < 前一个节点所有活动(包括课程和课间)时长之和** 的情况 _（如例图所示，上午节点开始时间与下午节点开始时间之差为 1h ，上午节点所有活动时长之和为 1.5h ， 1h < 1.5h ）_ ，如果出现此情况，则弹出错误弹窗提示用户修改时间线。
![例](https://github.com/user-attachments/assets/f1bb7ee7-6498-407d-9f31-ed813b4d2ef6)

### 最后一步

- [X] 我认为上述的描述已经足以详细，以便开发人员理解您想要的功能。如果我的 issue 没有按照上述的要求填写，可能会被无条件关闭。

### 不要乱下载别人发的文件，特别是名字是fix.zip的文件。您是否遇见过这样的文件？

否

**Comment by IsHPDuwu** [comment]
not planned

一周之内可能有多种时间线
